src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp

```diff
@@ -308,7 +308,11 @@ CTranslatorRelcacheToDXL::RetrieveRelIndexInfoForPartTable(CMemoryPool *mp,
 
 		GPOS_TRY
 		{
-			if (IsIndexSupported(index_rel))
+			// If the index is supported, but cannot yet be used, ignore it; but
+			// mark the plan we are generating and cache as transient.
+			// See src/backend/access/heap/README.HOT for discussion.
+			if (IsIndexSupported(index_rel) &&
+				!gpdb::MarkMDCacheAsTransient(index_rel))
```

Comment:
Could we move gpdb::MarkMDCacheAsTransient inside IsIndexSupported?

Reply:
IsIndexSupported is called in 2 places without gpdb::MarkMDCacheAsTransient. Why is it correct? If index is not visible, then we cannot use it, so the index is not supported

Reply:
may be rename IsIndexSupported to IsIndexSupportedAndUsable ?

Reply:
We only check for transient if the index is supported, right?

Reply:
if we combine the functions, we will throw an exception if the index cannot be used, which may not be valid.

Reply:
I don't think it's a good idea to combine these functions since there is an index type check. and if they are combined, it will be problematic to throw a correct exception.

Reply:
I have researched the other two uses of `IsIndexSupported`. The list of relationship indexes is obtained only in the `RetrieveRelIndexInfoForNonPartTable`, after which their `oid` are stored in the array in the relation object and information about indexes is obtained through `RetrieveIndex` during planning. I don't see a situation in which we can get an index in the `RetrieveIndex` that is not from the list built in the `RetrieveRelIndexInfoForNonPartTable`